datafusion/src/execution/dataframe_impl.rs

```diff
@@ -231,6 +231,29 @@ impl DataFrame for DataFrameImpl {
                 .build()?,
         )))
     }
+
+    fn intersect(&self, dataframe: Arc<dyn DataFrame>) -> Result<Arc<dyn DataFrame>> {
```

Comment:
looks like there is some code duplication between this and the one in sql planner, perhaps better to move this into logical plan builder so it can be reused in both places.